workflows/post_process_run/scream_post/format_output.py

```diff
@@ -0,0 +1,186 @@
+from typing import Hashable, List, Mapping
+import xarray as xr
+import numpy as np
+import os
+
+output_variables = [
+    "T_mid",
+    "qv",
+    "omega",
+    "horiz_winds",  # this will be changed to u and v in the future
+    "surf_sens_flux",
+    "surf_evap",
+    "SW_clrsky_flux_up@bot",
+    "SW_clrsky_flux_dn@bot",
+    "LW_clrsky_flux_dn@bot",
+    "LW_clrsky_flux_up@bot",
+    "SW_clrsky_flux_up@tom",
+    "LW_clrsky_flux_up@tom",
+    "VerticalLayerInterface",
+    "p_mid",
+    "ps",
+    "VerticalLayerInterface",
+    # "pseudo_density",  # this is pressure thickness
+    "area",
+    "lat",
+    "lon",
+    "hyai",
+    "hybi",
+]
+
+NP = 4
+NPG = 2
+levs = 128
+
+
+def make_placeholder_data(
+    sample: xr.DataArray, generate_variable: str, scaled_factor: float = 1.0
+):
+    placeholder = xr.DataArray(
+        np.random.rand(*sample.values.shape) * scaled_factor,
+        coords=sample.coords,
+        dims=sample.dims,
+    )
+    return placeholder.rename(generate_variable)
+
+
+def make_delp(p_mid, z_int, ps):
+    ptop = 225.52
+    delp = p_mid.copy()
+    dz = z_int[:, :, 0:-1] - z_int[:, :, 1:]
+    p_int = z_int.copy()
+    p_int[:, :, 0] = ptop
+    for k in range(1, levs):
+        p_int[:, :, k] = (
+            dz[:, :, k - 1] * p_mid[:, :, k] + dz[:, :, k] * p_mid[:, :, k - 1]
+        ) / (dz[:, :, k - 1] + dz[:, :, k])
+    p_int[:, :, -1] = ps / 2.0  # surface pressure bug
```

Comment:
Is this method more accurate than using `hyai`, `hybi`, and `ps` to compute the pressure on the layer interfaces, because SCREAM is a non-hydrostatic model?  It's too bad they don't have this output as a diagnostic directly...

More generally it would probably be good to have this written in more xarray-style code, e.g. without the assumption of array shapes and position of the vertical dimension.

Reply:
In the future, we should have `delp` directly as part of the output. Changed to using `hyai`, `hybi`, `p0`, and `ps` to calculate the full pressure.